### PR TITLE
simplify interface InstanceRoleProps

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

Extending multiple interfaces yields a simpler interface and consistent variable names.

## Does this change require changes to existing projects or CDK CLI?

No.

## How to test

Observe CI?

## How can we measure success?

Simpler code?

## Have we considered potential risks?

n/a

## Images

n/a

### CODE REVIEW

File: src/constructs/iam/policies/ssm.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export class GuSSMRunCommandPolicy extends GuPolicy {
     };
   }
 
-  constructor(scope: GuStack, id: string, props: GuPolicyProps) {
+  constructor(scope: GuStack, id: string = "SSMRunCommandPolicy", props: GuPolicyProps = { overrideId: false }) {
```
Comment (jamie-lynch) [inline src/constructs/iam/policies/ssm.ts:36]: I'm not sure that I like defining a default object for the prop and then also having a `getDefaultProps` function. It makes it a bit confusing to have two different places to define default props. If there are no required props, maybe the whole props parameter should be optional instead? 

```suggestion
  constructor(scope: GuStack, id: string = "SSMRunCommandPolicy", props?: GuPolicyProps) {
```

**Comment by jamie-lynch** [comment]
> Does this change require changes to existing projects or CDK CLI?
> No.

**Comment by jamie-lynch** [comment]
Clicked the wrong button there... 🤦

File: src/constructs/iam/policies/ssm.ts (hunk 1 above)
Comment (akash1810) [reply src/constructs/iam/policies/ssm.ts:36]: Yes, I think I agree with this, it gets us close to same shape as CDK classes too.

**Review by jamie-lynch — APPROVED** [review]
👍

**Pull request merged**